src/metaswitch/clearwater/etcd_shared/plugin_utils.py

```diff
@@ -52,23 +52,28 @@ def run_command(command, namespace=None, log_error=True):
     if namespace:
         command = "ip netns exec {} ".format(namespace) + command
 
-    try:
-        # Pass the close_fds argument to avoid the pidfile lock being held by
-        # child processes
-        output = subprocess.check_output(command,
-                                         shell=True,
-                                         stderr=subprocess.STDOUT,
-                                         close_fds=True)
-        _log.debug("Command {} succeeded and printed output {!r}".
-                   format(command, output))
+    # Pass the close_fds argument to avoid the pidfile lock being held by
+    # child processes
+
+    p = subprocess.Popen(command,
+                         shell=True,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE,
+                         close_fds=True)
+    stdout, stderr = p.communicate()
+    if p.returncode != 0:
+        # it failed, log the return code and output 
+        _log.error("Command {} failed with return code {}"
+                   " and printed output {!r}".format(command,
+                                                     p.returncode,
+                                                     p.output))
+        return p.returncode
+    else:
```

Comment:
(and you probably want a debug log if it is empty as well as the warning log if it isn't)